### PR TITLE
Adding in restify integration, docs, tests more...

### LINKED ISSUE #125: Write tests + documentation to explain how to use this package with restify
Followup to #85.

Apollo Server works _out-of-the-box_ with [Restify](http://restify.com/) 🎉 .

We still need to ensure the compatibility with it: we need to ensure it won't break anytime soon so we need to provide both tests and documentation for this!

### PR DESCRIPTION
**Pull request by joelgriffith**
Adds in restify, some new tests (would like to move these to a shared util and test in each integration). I've went through as much `md` files as I could to update fw support, but may have missed a place or two.

Let me know your thoughts!

Closes #125

### CODE REVIEW

**Comment by joelgriffith** [comment]
Appears coveralls isn't yet working. Here's the coverage stats after this PR:

```
=============================== Coverage summary ===============================
Statements   : 98.21% ( 660/672 )
Branches     : 89.46% ( 297/332 )
Functions    : 100% ( 151/151 )
Lines        : 99.06% ( 631/637 )
================================================================================```

**Comment by joelgriffith** [comment]
Hey folks, this has been sitting for a few weeks, any thoughts/concerns/comments?

**Comment by stubailo** [comment]
I'll take a look at it soon! Thanks for the reminder.

**Comment by helfer** [comment]
@joelgriffith Thanks, it looks good at first glance! What were the necessary changes compared to the express version? It might be helpful to point those out in comments in the code, because most of it looks very similar.

**Comment by helfer** [comment]
@joelgriffith let's get this merged some time soon! I think we're pretty close, so it would be a shame if we dropped it now.

**Comment by jadkap** [comment]
Hi.  Any idea when this PR might get merged?  I understand it's possible to use the express variant with restify, but it would be cleaner to have a restify variant out of the box.  I've pulled down @joelgriffith's fork to try locally (though running into a schema issue, but that's most likely because I'm new to GraphQL).  Thanks.

**Comment by DxCx** [comment]
Hi @jadkap, im happy to see you want to take over this,
if you need any help (running into schema issues for example ;)) feel free to consult on slack (\@hagaico)

**Comment by jadkap** [comment]
Haha @DxCx, did I say I wanted to take this over?  I will see, given time constraints.

**Comment by DxCx** [comment]
well @jadkap, you already downloaded it locally to try it off, if you make it work, why not contribute it back ;)
and if you didn't make it work, as i said, you are welcome to reach me for consultant over slack :)

**Comment by joelgriffith** [comment]
Hey folks, sorry, been tied up as does seem the case.

I'm lost on where this is... I only see a conflict in CHANGELOG.md, I can fix whatever and push?

**Comment by DxCx** [comment]
@joelgriffith , you better hold on for #180 which will be merged any time now,
after you will rebase upon it, your integration code will look much much smaller.

File: README.md [hunk 1]
```diff
@@ -122,6 +122,26 @@ app.use(router.allowedMethods());
...
+
+const server = restify.createServer({
+  title: 'GraphQL Server'
+});
+
+server.use(restify.bodyParser());
+
+server.post('/graphql', graphqlRestify({ schema: myGraphQLSchema }));
+
+sever.get('/graphiql', graphiqlRestify({ endpointURL: '/graphql' }));
```
Comment (jadkap) [inline README.md]: fix typo

File: packages/graphql-server-restify/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "graphql-server-module-graphiql": "^0.4.3"
+  },
+  "devDependencies": {
+    "@types/restify": "^2.0.33",
+    "graphql-server-integration-testsuite": "^0.4.3",
+    "restify": "^4.1.1",
+    "typed-graphql": "^1.0.2"
+  },
+  "peerDependencies": {
+    "graphql": "^0.6.1 || ^0.7.0"
```
Comment (jadkap) [inline packages/graphql-server-restify/package.json]: add || ^0.8.0

**Comment by DxCx** [comment]
0.5.1 branch is merged :)
@jadkap / @joelgriffith can one of you take over and rebase / align so we can merge?

File: README.md [hunk 3]
```diff
@@ -122,6 +122,26 @@ app.use(router.allowedMethods());
...
+
+const PORT = 3000;
+
+const server = restify.createServer({
+  title: 'GraphQL Server'
+});
+
+server.use(restify.bodyParser());
+
+server.post('/graphql', graphqlRestify({ schema: myGraphQLSchema }));
```
Comment (DxCx) [inline README.md]: should be `use` as now apollo supports GET requests as well. :)

**Comment by joelgriffith** [comment]
I'll take a stab at it later today, will push once everything is resolved

**Comment by DxCx** [comment]
great @joelgriffith thanks! 👍

**Comment by joelgriffith** [comment]
Pushed my rebase and fixes. I'm not able to complete an install or run tests, so I'm not sure what the issue is:

```
# jgriffith @ jgriffith in ~/Griffin/graphql-server on git:feature/restify-integration|MERGING x [11:46:51] C:1
$ npm i

> undefined postinstall /Users/jgriffith/Griffin/graphql-server
> lerna bootstrap

Lerna v2.0.0-beta.30
Bootstrapping 8 packages
Installing external dependencies
Errored while running BootstrapCommand.execute
npm ERR! Darwin 15.6.0
npm ERR! argv "/Users/jgriffith/.nvm/versions/node/v6.9.1/bin/node" "/Users/jgriffith/.nvm/versions/node/v6.9.1/bin/npm" "install" "@types/restify@^2.0.33" "graphql-server-integration-testsuite@^0.4.3" "restify@^4.1.1" "typed-graphql@^1.0.2" "graphql-server-core@^0.4.3"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code E404

npm ERR! 404 Registry returned 404 for GET on https://registry.npmjs.org/graphql-server-integration-testsuite
npm ERR! 404
npm ERR! 404  'graphql-server-integration-testsuite' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/jgriffith/Griffin/graphql-server/packages/graphql-server-restify/npm-debug.log
```

File: packages/graphql-server-restify/src/restifyApollo.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      }
+
+      res.statusCode = error.statusCode;
+      res.write(error.message);
+      res.end();
+    });
+  };
+}
+
+function isOptionsFunction(arg: GraphQLOptions | RestifyGraphQLOptionsFunction): arg is RestifyGraphQLOptionsFunction {
```
Comment (DxCx) [inline packages/graphql-server-restify/src/restifyApollo.ts]: I think this function here can be removed
It moved to core.. please check that ;)

File: packages/graphql-server-restify/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "rootDir": "./src",
+    "outDir": "./dist",
+    "allowSyntheticDefaultImports": false,
+    "pretty": true,
+    "removeComments": true,
+    "typeRoots": [
+        "node_modules/@types"
+    ],
+    "types": [
+        "typed-graphql",
```
Comment (DxCx) [inline packages/graphql-server-restify/tsconfig.json]: Should be removed

File: packages/graphql-server-restify/package.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  "homepage": "https://github.com/apollostack/graphql-server#readme",
+  "dependencies": {
+    "graphql-server-core": "^0.4.3",
+    "graphql-server-module-graphiql": "^0.4.3"
+  },
+  "devDependencies": {
+    "@types/restify": "^2.0.33",
+    "graphql-server-integration-testsuite": "^0.4.3",
+    "restify": "^4.1.1",
+    "typed-graphql": "^1.0.2"
```
Comment (DxCx) [inline packages/graphql-server-restify/package.json]: Should be replaced with @types/graphql

File: packages/graphql-server-restify/package.json [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  "peerDependencies": {
+    "graphql": "^0.6.1 || ^0.7.0 || ^0.8.0"
+  },
+  "optionalDependencies": {
+    "@types/restify": "^2.0.33"
```
Comment (DxCx) [inline packages/graphql-server-restify/package.json]: Should add @types/graphql

File: packages/graphql-server-restify/src/restifyApollo.test.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+  options.graphqlOptions = options.graphqlOptions || { schema: Schema };
+  if (!options.excludeParser) {
+    server.use(restify.bodyParser());
+  }
+  if (options.graphiqlOptions ) {
+    server.get('/graphiql', graphiqlRestify( options.graphiqlOptions ));
+  }
+  server.post('/graphql', graphqlRestify( options.graphqlOptions ));
+  server.put('/graphql', graphqlRestify( options.graphqlOptions ));
```
Comment (DxCx) [inline packages/graphql-server-restify/src/restifyApollo.test.ts]: Why put and not use?

**Comment by joelgriffith** [comment]
OK! Types are fixed, as are tests, and README. The interesting thing to note is that restify doesn't support namespacing of the `.use` API, so you have to verbosely do a `.get` and a `.post` for query parsing and json parsing to work. README shows correct usage (and tests do as well).

File: packages/graphql-server-restify/src/restifyApollo.test.ts (hunk 8 above)
Comment (joelgriffith) [inline packages/graphql-server-restify/src/restifyApollo.test.ts]: Can't namespace a `.use` call: https://github.com/restify/node-restify/issues/289

**Comment by DxCx** [comment]
@helfer please merge #274 which should fix the travis issues,
then merge this upon this should give you a branch that passes tests.

**Comment by helfer** [comment]
Thanks @joelgriffith for fixing it up, and thanks @DxCx for fixing the travis issues. You guys are awesome!

**Comment by jadkap** [comment]
Thanks all for getting this merged.  I was able to try latest and found a couple of things.  (BTW, the "schema" issue I previously encountered was a symlink/node_modules issue.)  @joelgriffith, if you have time, I'd appreciate your input on https://github.com/apollostack/graphql-server/pull/285.

**Pull request merged**